lib/commands/general.js

```diff
@@ -47,6 +47,19 @@ commands.getWindowSize = async function () {
   return await this.uiautomator2.jwproxy.command('/window/current/size', 'GET', {});
 };
 
+commands.execute = async function (script, args) {
+  if (script.match(/^mobile\:/)) {
+    script = script.replace(/^mobile\:/, '').trim();
+    if (script === 'scroll' && !this.isWebContext()) {
```

Comment:
These conditionals seem weird. We can dispatch with the web context in one place and then just work on the value of `script`:
```js
if (this.isWebContext()) {
  // proxy to chromedriver
  return;
}
if (script.match(/^mobile\:/)) {
  let mobileScript = script.replace(/^mobile\:/, '').trim();
  if (mobileScript === 'scroll') { // why 'scroll' and not 'scrollTo'?
    let elementId = args[0].element;
    return await this.uiautomator2.jwproxy.command(`/appium/element/${elementId}/scroll_to`, 'POST', {});
  } else if (mobileScript === 'viewportScreenshot') {
    return await this.getViewportScreenshot();
  }
}

throw new Error(`Unable to execute unknown script: '${script}'`);
```